### PR TITLE
test: check rgw pod is running for canary github workflow

### LINKED ISSUE #11743: The encryption-pvc-kms-vault-token canary test is failing frequently
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
As pointed out in #11741, the encryption-pvc-kms-vault-token test is failing about half the time in master.

The failure is seen during the step [validate rgw vault transit](https://github.com/rook/rook/actions/runs/4234791073/jobs/7357581690):
```
++ kubectl -n rook-ceph exec rook-ceph-rgw-my-store-a-568c9dcdb5-cwjgj -- cat /etc/vault/rgw/ssekms/vault.token
Defaulted container "rgw" out of: rgw, chown-container-data-dir (init), vault-initcontainer-token-file-setup (init)
Error from server: 
+ VAULT_TOKEN=
Error: Process completed with exit code 1.
```

Or just before the token query, another failure:
```
No resources found in rook-ceph namespace.
++ kubectl -n rook-ceph get pods -l app=rook-ceph-rgw
++ awk 'FNR == 2 {print $1}'
+ RGW_POD=
++ kubectl -n rook-ceph describe pods ''
++ grep rgw-crypt-vault-token-file
++ cut -f2- -d=
error: resource name may not be empty
+ RGW_TOKEN_FILE=
```

A successful run of this command shows the token was successfully retrieved:
```
++ kubectl -n rook-ceph exec rook-ceph-rgw-my-store-a-56b7664fd5-jh24h -- cat /etc/vault/rgw/ssekms/vault.token
Defaulted container "rgw" out of: rgw, chown-container-data-dir (init), vault-initcontainer-token-file-setup (init)
+ VAULT_TOKEN=hvs.CAESIAmxzyMzeh-aAIUGV6j8-rxz33_FSAqZbVhZMRidzU_zGh4KHGh2cy51d0RrQzFtRkNKRXV1eTU4ZzcwZ0U3dlY
```

**Expected behavior:**
Passing tests

### PR DESCRIPTION
**Pull request by thotz**
**Description of your changes:**
For the RGW daemon validation please check whether pod is Running than the existing checks

**Which issue is resolved by this Pull Request:**
Resolves #11743

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure)).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by travisn** [comment]
Looks like the test is still failing.
```
++ cut -f2- -d=
++ grep rgw-crypt-vault-token-file
++ kubectl -n rook-ceph describe pods ''
error: resource name may not be empty
+ RGW_TOKEN_FILE=
Error: Process completed with exit code 1.
```

File: tests/scripts/validate_cluster.sh [hunk 1]
```diff
@@ -62,8 +62,12 @@ function test_demo_osd {
 }
 
 function test_demo_rgw {
-  # shellcheck disable=SC2046
-  return $(wait_for_daemon "$EXEC_COMMAND -s | grep -sq 'rgw:'")
```
Comment (BlaineEXE) [inline tests/scripts/validate_cluster.sh:66]: Should we consider updating `wait_for_daemon` to check for `Running`? @travisn ?

Comment (travisn) [reply tests/scripts/validate_cluster.sh:66]: Lookks good now waiting for the pod to be ready

**Review by travisn — APPROVED** [review]

File: tests/scripts/validate_cluster.sh (hunk 1 above)
Comment (thotz) [reply tests/scripts/validate_cluster.sh:66]: `Status.Phase` is Running even when Pod is in terminating status and also even when before all the contianers are ready

**Pull request merged**